ext/hal/st/stm32cube/CMakeLists.txt

```diff
@@ -28,6 +28,8 @@ if(CONFIG_SOC_SERIES_STM32F0X)
   zephyr_sources(stm32f0xx/drivers/src/stm32f0xx_hal.c)
   zephyr_sources(stm32f0xx/drivers/src/stm32f0xx_hal_rcc.c)
   zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_STM32_CUBE stm32f0xx/drivers/src/stm32f0xx_ll_utils.c)
+  zephyr_sources_ifdef(CONFIG_I2C stm32f0xx/drivers/src/stm32f0xx_ll_i2c.c)
```

Comment:
Shouldn´t that be CONFIG_I2C_STM32_V2 instead of CONFIG_I2C?

SPI uses CONFIG_SPI_STM32 <-> I2C uses CONFIG_I2C. By using just CONFIG_I2C there may be a situation where stm32f0xx_ll_i2c.c is added to the sources but the header file is excluded from soc.h (e.g. in case another variant of the I2C driver is used).

Reply:
You are right, will fix